### PR TITLE
Add doukutsu-save-editor-git

### LINKED ISSUE #3640: [打包请求] doukutsu-save-editor-git
### 问题类型 / Type of issues

* 软件打包请求 / package request

----

* 简述 / Brief Description

    [doukutsu-rs](https://github.com/doukutsu-rs/doukutsu-rs) 的能跑 240fps 的存档修改器

* 主要语言和工具链 / Main Languages and Toolchains

    Rust; egui (GTK3)

* 上游链接 / Upstream Link

    https://aur.archlinux.org/packages/doukutsu-save-editor

* 为何需要这个软件包 / Why Need This Package

    洞穴物语的Linux原生修改器
    现有跨平台实现多为 java

### PR DESCRIPTION
**Pull request by poly000**
close #3640

### CODE REVIEW

File: archlinuxcn/doukutsu-save-editor-git/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+pkgver=r63.gb26926f
+pkgrel=1
+pkgdesc="CaveStory (doukutsu) save editor"
+url="https://github.com/poly000/${_pkgname}-rs"
+conflicts=('doukutsu-save-editor')
+provides=('doukutsu-save-editor')
+arch=('x86_64')
+license=('MIT')
+depends=('libxcb' 'libspeechd' 'libxkbcommon' 'gtk3' 'at-spi2-core')
+makedepends=('cargo' 'openssl')
```
Comment (Zenvie) [inline archlinuxcn/doukutsu-save-editor-git/PKGBUILD:14]:
```
==> ERROR: Cannot find the git package needed to handle git sources.
```

File: archlinuxcn/doukutsu-save-editor-git/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    echo "r$(git rev-list --count HEAD).g$(git rev-parse --short HEAD)"
+}
+
+build() {
+    cd "${_pkgname}-rs"
+
+    export CARGO_TARGET_DIR=target
+
+    cargo build --frozen --release
+}
```
Comment (Zenvie) [inline archlinuxcn/doukutsu-save-editor-git/PKGBUILD:39]:
```
doukutsu-save-editor-git W: ELF file ('usr/bin/doukutsu-save-editor') lacks GNU_PROPERTY_X86_FEATURE_1_SHSTK.
```

File: archlinuxcn/doukutsu-save-editor-git/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+pkgver=r63.gb26926f
+pkgrel=1
+pkgdesc="CaveStory (doukutsu) save editor"
+url="https://github.com/poly000/${_pkgname}-rs"
+conflicts=('doukutsu-save-editor')
+provides=('doukutsu-save-editor')
+arch=('x86_64')
+license=('MIT')
+depends=('libxcb' 'libspeechd' 'libxkbcommon' 'gtk3' 'at-spi2-core')
+makedepends=('cargo' 'openssl' 'git')
```
Comment (Zenvie) [inline archlinuxcn/doukutsu-save-editor-git/PKGBUILD:14]:
```
doukutsu-save-editor-git W: Dependency glibc detected and implicitly satisfied (libraries ['usr/lib/libc.so.6', 'usr/lib/libm.so.6'] needed in files ['usr/bin/doukutsu-save-editor'])
doukutsu-save-editor-git W: Dependency gcc-libs detected and implicitly satisfied (libraries ['usr/lib/libgcc_s.so.1'] needed in files ['usr/bin/doukutsu-save-editor'])
doukutsu-save-editor-git W: Dependency glib2 detected and implicitly satisfied (libraries ['usr/lib/libglib-2.0.so.0', 'usr/lib/libgobject-2.0.so.0'] needed in files ['usr/bin/doukutsu-save-editor'])
doukutsu-save-editor-git W: Dependency included, but may not be needed ('libxcb')
doukutsu-save-editor-git W: Dependency included, but may not be needed ('libspeechd')
doukutsu-save-editor-git W: Dependency included, but may not be needed ('libxkbcommon')
doukutsu-save-editor-git W: Dependency included, but may not be needed ('at-spi2-core')
```

File: archlinuxcn/doukutsu-save-editor-git/PKGBUILD (hunk 2 above)
Comment (poly000) [reply archlinuxcn/doukutsu-save-editor-git/PKGBUILD:50]: 这是栈保护机制？

Comment (Zenvie) [reply archlinuxcn/doukutsu-save-editor-git/PKGBUILD:50]: I don’t know how to resolve this warning message, but if the software is working fine, you can ignore it for now.

**Pull request merged**